### PR TITLE
Refactors case statistical methods

### PR DESCRIPTION
**Pull request by rlgreen91**
Directly addresses Issue #1301. Refactors methods for calculating percentage growth statistics for cases.  Moves statistical methods for cases to separate statistics module.  Refactors analytics controller and dashboard using statistics module.

To be clear, there is a lot to do with regard to refactoring these methods that appear in multiple models.  On top of that, there are enhancements we could do like using value objects or performing some of these calculations at the database level.  But, this is a good first step.

In your PR did you:

  - [X] Include a description of the changes?
  - [X] Mention the issue the PR addresses?
  - [ ] Include screenshots of any changes to the UI?
  - [ ] Isolate any changes to gems (meaning that any new, updated, or removed gems and resulting code changes should be in their own PR)?
  - [X] Add and/or update specs for your code?

### CODE REVIEW

File: spec/lib/statistics_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Statistics do
+
+  def mom_changed_cases_growth(start_cases:, end_cases:)
```
Comment (trystant) [inline spec/lib/statistics_spec.rb]: Seems like these could be helpful in the `Statistics` model itself, not just in the specs

File: spec/lib/statistics_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+require 'rails_helper'
+
+RSpec.describe Statistics do
+
+  def mom_changed_cases_growth(start_cases:, end_cases:)
+    Statistics.mom_changed_cases_growth(cases_at_start: start_cases, cases_at_end: end_cases)
+  end
+
+  def mom_cases_growth(start_cases:, end_cases:)
+    Statistics.mom_cases_growth(cases_at_start: start_cases, cases_at_end: end_cases)
```
Comment (trystant) [inline spec/lib/statistics_spec.rb]: What happens if an invalid date or other data types are sent as arguments?

File: spec/lib/statistics_spec.rb (hunk 1 above)
Comment (rlgreen91) [reply spec/lib/statistics_spec.rb]: What, to basically have instance methods that act as a shell for class methods.  I'm not so sure if that's a good idea - I like being explicit about what module the method is coming from.

File: spec/lib/statistics_spec.rb (hunk 2 above)
Comment (rlgreen91) [reply spec/lib/statistics_spec.rb]: Yeah, I have that as one of my things to tackle as part of this.  It's technically always been a concern, but everything was so hard coded that we didn't have to care.

**Comment by rlgreen91** [comment]
@trystant I can't get rid of that carriage return it's crying about but otherwise it's good.

**Comment by trystant** [comment]
@rlgreen91  You tried `rubocop -a`?

**Comment by rlgreen91** [comment]
I'm worried it's going to change all of the files, and I'll have to clean that up from a git standpoint.

**Comment by trystant** [comment]
You can mention `rubocop -a <filename>` with a filename so it doesn't do the whole repo

**Pull request merged**